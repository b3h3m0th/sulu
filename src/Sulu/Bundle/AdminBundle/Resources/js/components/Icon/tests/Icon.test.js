// @flow
import React from 'react';
import {createEvent, fireEvent, render, screen} from '@testing-library/react';
import log from 'loglevel';
import Icon from '../Icon';

jest.mock('loglevel', () => ({
    warn: jest.fn(),
}));

test('Icon should render', () => {
    const {container} = render(<Icon name="su-save" />);
    expect(container).toMatchSnapshot();
});

test('Icon should not render with invalid icon', () => {
    const {container} = render(<Icon name="xxx" />);
    expect(container).toMatchSnapshot();
    expect(log.warn).toHaveBeenCalled();
});

test('Icon should not render with empty string', () => {
    const {container} = render(<Icon name="" />);
    expect(container).toMatchSnapshot();
    expect(log.warn).toHaveBeenCalled();
});

test('Icon should render with class names', () => {
    const {container} = render(<Icon className="test" name="su-pen" />);
    expect(container).toMatchSnapshot();
});

test('Icon should render with onClick handler, role and tabindex', () => {
    const onClickSpy = jest.fn();
    const {container} = render(<Icon className="test" name="su-save" onClick={onClickSpy} />);
    expect(container).toMatchSnapshot();
});

test('Icon should call the callback on click', () => {
    const onClick = jest.fn();
    const stopPropagation = jest.fn();
    render(<Icon className="test" name="su-pen" onClick={onClick} />);
    const icon = screen.queryByLabelText('su-pen');

    const clickEvent = createEvent.click(icon, {stopPropagation});
    clickEvent.stopPropagation = stopPropagation;
    fireEvent(icon, clickEvent);
    console.log(clickEvent);
    expect(clickEvent.stopPropagation).toBeCalled();
    expect(onClick).toBeCalled();
});

// test('Icon should call the callback on when space is pressed', () => {
//     const onClick = jest.fn();
//     const stopPropagation = jest.fn();
//     const icon = shallow(<Icon className="test" name="su-pen" onClick={onClick} />);
//     icon.simulate('keypress', {key: ' ', stopPropagation});
//     expect(onClick).toBeCalled();
//     expect(stopPropagation).toBeCalled();
// });

// test('Icon should call the callback on when enter is pressed', () => {
//     const onClick = jest.fn();
//     const stopPropagation = jest.fn();
//     const icon = shallow(<Icon className="test" name="su-pen" onClick={onClick} />);
//     icon.simulate('keypress', {key: 'Enter', stopPropagation});
//     expect(onClick).toBeCalled();
//     expect(stopPropagation).toBeCalled();
// });
