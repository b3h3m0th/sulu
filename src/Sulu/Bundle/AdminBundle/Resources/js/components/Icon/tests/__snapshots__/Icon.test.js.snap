// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should not render with empty string 1`] = `<div />`;

exports[`Icon should not render with invalid icon 1`] = `<div />`;

exports[`Icon should render 1`] = `
<div>
  <span
    aria-label="su-save"
    class="su-save"
  />
</div>
`;

exports[`Icon should render with class names 1`] = `
<div>
  <span
    aria-label="su-pen"
    class="su-pen test"
  />
</div>
`;

exports[`Icon should render with onClick handler, role and tabindex 1`] = `
<div>
  <span
    aria-label="su-save"
    class="su-save clickable test"
    role="button"
    tabindex="0"
  />
</div>
`;
